### ISSUE TITLE
请问trainer.py中SOC loss定义是不是有问题

### ISSUE THREAD

**Issue by StarkerRegen** [ZHKKKe/MODNet#111]
```
backup_matte_loss = boundaries * F.l1_loss(pred_matte, pred_backup_matte, reduction='none') 
backup_matte_loss = torch.sum(backup_matte_loss, dim=(1,2,3)) / torch.sum(boundaries, dim=(1,2,3))    
backup_matte_loss = torch.mean(backup_matte_loss)
```
作者您好，上述代码是trainer.py中SOC函数对L<sub>cons</sub>的部分定义（284行~287行）。在我看来，这里相当于m<sub>d</sub>*L1(α<sub>p</sub>-α'<sub>p</sub>)，但论文中描述的是m<sub>d</sub>*L1(α<sub>p</sub>-d<sub>p</sub>)。

**Comment by czHappy** [comment]
I have checked the code before, and now I have the same question. I think the loss md*L1(αp-α'p) is bizarre, because it seems that the loss would  make SOC model to degenerate to the original model, maybe......

**Comment by ZHKKKe** [comment]
Hi, both. 

Thanks for your attention.

Following is a comment in the code:
https://github.com/ZHKKKe/MODNet/blob/a1ea72fc85b5314bd820ae00e37ea05a04401004/src/trainer.py#L278
The provided code has similar results as the formula in the paper. You can also try to implement the code the same as the paper. We will update the formula in the next version of paper.

@czHappy For your question, the backup model produced $\alpha^{'}_{p}$ is repleaced by the latest model after each SOC epoch. Hence, the model is not degenerated.
